src/prpy/base/robot.py

```diff
@@ -284,15 +284,33 @@ def ViolatesVelocityLimits(self, traj):
         return False
 
     def _PlanWrapper(self, planning_method, args, kw_args):
-        # Call the planner.
-        traj = planning_method(self, *args, **kw_args)
 
-        # Strip inactive DOFs from the trajectory.
+        # Call the planner.
         config_spec = self.GetActiveConfigurationSpecification()
-        openravepy.planningutils.ConvertTrajectorySpecification(traj, config_spec)
+        result = planning_method(self, *args, **kw_args)
+
+        # Define the post processing steps for the trajectory.
+        def postprocess_trajectory(traj, kw_args):
+
+            # Strip inactive DOFs from the trajectory.
+            openravepy.planningutils.ConvertTrajectorySpecification(
+                traj, config_spec
+            )
+
+            # Optionally execute the trajectory.
+            if 'execute' not in kw_args or kw_args['execute']:
+                return self.ExecuteTrajectory(traj, **kw_args)
+            else:
+                return traj
```

Comment:
Re: your latest comment.

What do you mean by "as it was executed"? Currently, we return the trajectory that was passed to the controller. This only differs from the input trajectory if the input was modified by the pre-processing steps (e.g. smoothed or re-timed). It does **not** reflect the actual trajectory that was followed by the manipulator.

For example, the a move-until-touch trajectory returned by `ExecuteTrajectory` will still be the full-length trajectory. Even if the actual trajectory executed on the robot immediately terminated due to an observed force.